src/components/items.tsx

```diff
@@ -5,43 +5,81 @@ import { v4 as uuidv4 } from "uuid";
 
 export function Items() {
   const [input, setInput] = useState("");
-  const [things, setThings] = useState<TItemDocument[]>([]);
-  useEffect(() => {
+  const [things, setThings] = useState<Partial<TItemDocument>[]>([]);
+
+  const fetchItems = () =>
     itemsServices.listItems().then((res) => {
       setThings(res);
     });
+  useEffect(() => {
+    fetchItems();
   }, []);
   const addItem = () => {
     const $id = uuidv4();
     const newItem = {
       name: input,
       isSelected: false,
     };
+    setThings((prev) => [...prev, { $id, ...newItem }]);
+
     itemsServices.newItem($id, newItem);
     setInput("");
   };
+  const updateItem = (t: Partial<TItemDocument>) => {
+    t.isSelected = !t.isSelected!;
+    const newArray = things.filter((i) => i.$id !== t.$id);
+    const index = things.findIndex((i) => i.$id === t.$id);
+    newArray.splice(index, 0, t);
```

Comment:
It'd be safer to check first if the index exists.
```suggestion
    if(index !== -1) {
        newArray.splice(index, 0, t);
    }
```